src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -2645,11 +2736,20 @@ void Pane::_AdvanceSnappedDimension(const bool widthOrHeight, LayoutSizeNode& si
         }
         else
         {
-            const auto cellSize = _control.CharacterDimensions();
+            const auto cellSize = termControl.CharacterDimensions();
             sizeNode.size += widthOrHeight ? cellSize.Width : cellSize.Height;
         }
     }
-    else
+    else if (_IsLeaf())
+    {
+        // If we're a leaf that didn't have a TermControl, then just increment
+        // by one. We have to increment by _some_ value, because this is used in
+        // a while() loop to find the next bigger size we can snap to. But since
+        // a non-terminal control doesn't really care what size it's snapped to,
+        // we can just say "one pixel larger is the next snap point"
+        sizeNode.size += 1;
+    }
+    else if (!_IsLeaf())

```

Comment:
This is another bit of weird conditionalness. Shouldn't you just outer check if (_IsLeaf()) then inner that check if (termControl) and else... and then leave the final outer else here for !_IsLeaf() instead of testing it again?

Reply:
Yea probably.